polkadot/zombienet-sdk-tests/tests/functional/validator_disabling.rs

```diff
@@ -0,0 +1,142 @@
+						.with_args(vec![
+							"-lMALUS=trace".into(),
+							// Without this the malus validator won't run on macOS.
+							"--insecure-validator-i-know-what-i-do".into(),
+						])
+						// Make it vulenrable so disabling really happens
+						.invulnerable(false)
+				});
+			// Also honest validators.
+			let r = (0..3).fold(r, |acc, i| {
+				acc.with_node(|node| {
+					node.with_name(&format!("honest-validator-{i}"))
+						.with_args(vec![("-lparachain=debug,runtime::staking=debug".into())])
+				})
+			});
+			r
+		})
+		.with_parachain(|p| {
+			p.with_id(1000)
+				.with_default_command("adder-collator")
+				.cumulus_based(false)
+				.with_default_image(images.cumulus.as_str())
+				.with_default_args(vec!["-lparachain=debug".into()])
+				.with_collator(|n| n.with_name("alice"))
+		})
+		.build()
+		.map_err(|e| {
+			let errors = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errors: {errors}")
+		})?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	log::info!("Spawning network");
+	let network = spawn_fn(config_builder).await?;
+
+	log::info!("Waiting for parablocks to be produced");
+	let honest_validator = network.get_node("honest-validator-0")?;
+	let relay_client: OnlineClient<PolkadotConfig> = honest_validator.wait_client().await?;
+
+	assert_finalized_para_throughput(
+		&relay_client,
+		20,
+		[(polkadot_primitives::Id::from(1000), 10..30)].into_iter().collect(),
+	)
+	.await?;
+
+	log::info!("Wait for a dispute to be initialized.");
+	let mut best_blocks = relay_client.blocks().subscribe_best().await?;
+	let mut dispute_session: u32 = u32::MAX;
+	// Check next new block from the current best fork
+	while let Some(block) = best_blocks.next().await {
+		let disputes = relay_client
+			.runtime_api()
+			.at(block?.hash())
+			.call_raw::<Vec<(SessionIndex, CandidateHash, DisputeState<BlockNumber>)>>(
+				"ParachainHost_disputes",
+				None,
+			)
+			.await?;
+		if let Some((session, _, _)) = disputes.first() {
+			dispute_session = *session;
+			break;
+		}
+	}
+	assert_ne!(dispute_session, u32::MAX);
+	log::info!("Dispute initiated.");
+
+	let concluded_dispute_metric =
+		"polkadot_parachain_candidate_dispute_concluded{validity=\"invalid\"}";
+	let parachain_candidate_dispute_metric = "parachain_candidate_disputes_total";
+	// honest-validator-1: reports parachain_candidate_disputes_total is at least 1 within 600
+	// seconds
+	honest_validator
+		.wait_metric_with_timeout(parachain_candidate_dispute_metric, |d| d >= 1.0, 600_u64)
+		.await?;
+	// honest-validator: reports polkadot_parachain_candidate_dispute_concluded{validity="invalid"}
+	// is at least 1 within 200 seconds
+	honest_validator
+		.wait_metric_with_timeout(concluded_dispute_metric, |d| d >= 1.0, 200_u64)
+		.await?;
+	// honest-validator: log line contains "Disabled validators detected" within 180 seconds
+	let result = honest_validator
+		.wait_log_line_count_with_timeout(
```

Comment:
Not sure if this is desired to check the logs, but why don't you read `Session::DisabledValidators` storage items?